storage/txcache/txCache.go

```diff
@@ -234,6 +234,22 @@ func (cache *TxCache) GetTransactionsPoolForSender(sender string) [][]byte {
 	return txsHashes
 }
 
+// GetLastPoolNonceForSender returns the last nonce from pool for the sender
+func (cache *TxCache) GetLastPoolNonceForSender(sender string) (uint64, bool) {
+	listForSender, ok := cache.txListBySender.getListForSender(sender)
+	if !ok {
+		return 0, ok
+	}
+
+	lastTxElement := listForSender.items.Back()
+	lastTx, ok := lastTxElement.Value.(*WrappedTransaction)
+	if !ok {
+		return 0, ok
+	}
+
+	return lastTx.Tx.GetNonce(), ok
```

Comment:
are the transactions in order by nonce? if no, this code doesn't cover all the situations.
If I send txs with nonces 1, 2, 3, 4, 7, 6, 5 in this order. will you code return 7 or 5 ?

Reply:
the structure that holds the txs by sender keeps them sorted by nonce.. [this unittest](https://github.com/ElrondNetwork/elrond-go/pull/4232/files#diff-c6a182d4b2fde8b5763c5498e76fa7e8b1e1c32e3acfe27aac2284c4a6b43d1bR285) should cover exactly this situation